internal/core/src/segcore/ConcurrentVector.h

```diff
@@ -144,16 +266,26 @@ class ConcurrentVectorImpl : public VectorBase {
     }
 
     void
-    set_data_raw(ssize_t element_offset, const void* source, ssize_t element_count) override {
-        set_data(element_offset, static_cast<const Type*>(source), element_count);
+    fill_chunk_data(const void* source, ssize_t element_count) override {
+        if (element_count == 0) {
+            return;
+        }
+        AssertInfo(chunks_.size() == 0, "no empty concurrent vector");
+        chunks_.emplace_to_at_least(1, Dim * element_count);
+        set_data(0, static_cast<const Type*>(source), element_count);
     }
 
     void
-    set_data(ssize_t element_offset, const Type* source, ssize_t element_count) {
+    set_data_raw(ssize_t element_offset, const void* source, ssize_t element_count) override {
```

Comment:
move all implementation to .cpp

Reply:
This is the member function of the template class